proxy/proxy.go

```diff
@@ -328,6 +330,7 @@ type Proxy struct {
 	defaultHTTPStatus        int
 	routing                  *routing.Routing
 	registry                 *routing.EndpointRegistry
+	rnd                      *rand.Rand
```

Comment:
This is only used by fadeIn but has to be passed across all proxy codepath.
I think it would be better if we wrap fadeIn logic (and encapsulate `rnd` by it) into custom type and put it into a separate file.
We can also make selectEndpoint and mapRequest and makeBackedRequest members of proxy such that we dont need to pass fadeIn logic as an argument.

Reply:
> into custom type
We still would need to keep `registry` inside `type Proxy struct`, so the only thing we could do is to replace `rnd` with this custom type. However, this won't decrease the complexity of `type Proxy struct` imho, so I decided not to do this.

> We can also make selectEndpoint and mapRequest and makeBackedRequest members of proxy
Yes, I like this idea very much, I've done this.